### ISSUE TITLE
CURL request when running 'cog run', 'cog build', or any cog-related commands fail (Cannot use Cog)

### ISSUE THREAD

**Issue by blevlabs** [replicate/cog#1215]
Hello, so I am following the guide on the replicate site for using [Cog to push a model to replicate](https://replicate.com/docs/guides/push-a-model), but every time I try to use Cog as shown, I get the following:

```
 => ERROR [stage-1  4/10] RUN curl -s -S -L https://raw.githubusercontent.com/pyenv/pyenv-installer/mast  1.1s
------                                                                                                         
 > [stage-1  4/10] RUN curl -s -S -L https://raw.githubusercontent.com/pyenv/pyenv-installer/master/bin/pyenv-installer | bash &&      git clone https://github.com/momo-lab/pyenv-install-latest.git "$(pyenv root)"/plugins/pyenv-install-latest &&         pyenv install-latest "3.11" &&  pyenv global $(pyenv install-latest --print "3.11") &&         pip install "wheel<1":                                                                         
0.370 curl: (60) SSL: no alternative certificate subject name matches target host name 'raw.githubusercontent.com'
0.370 More details here: https://curl.se/docs/sslcerts.html
0.370 
0.370 curl failed to verify the legitimacy of the server and therefore could not
0.370 establish a secure connection to it. To learn more about this situation and
0.370 how to fix it, please visit the web page mentioned above.
0.371 /bin/sh: 1: pyenv: not found
0.372 Cloning into '/plugins/pyenv-install-latest'...
0.963 /bin/sh: 1: pyenv: not found
------
```

How can I resolve this issue? I would like to start uploading my models to Replicate, but this verification issue is making that impossible

**Comment by voodoohop** [comment]
Having exactly the same issue.

**Comment by daanelson** [comment]
hey @blevlabs @voodoohop @isouf, believe this is fixed now. Give it a try and let us know if you're still seeing problems.

**Comment by blevlabs** [comment]
@daanelson I reinstalled Cog and still get the following:
```
 => CACHED [stage-1  3/12] RUN --mount=type=cache,target=/var/cache/apt apt-get update -qq && apt-get in  0.0s
 => ERROR [stage-1  4/12] RUN curl -s -S -L https://raw.githubusercontent.com/pyenv/pyenv-installer/mast  0.7s
------                                                                                                         
 > [stage-1  4/12] RUN curl -s -S -L https://raw.githubusercontent.com/pyenv/pyenv-installer/master/bin/pyenv-installer | bash &&      git clone https://github.com/momo-lab/pyenv-install-latest.git "$(pyenv root)"/plugins/pyenv-install-latest &&         pyenv install-latest "3.11" &&  pyenv global $(pyenv install-latest --print "3.11") &&         pip install "wheel<1":                                                                         
0.290 curl: (60) SSL: no alternative certificate subject name matches target host name 'raw.githubusercontent.com'
0.290 More details here: https://curl.se/docs/sslcerts.html
0.290 
0.290 curl failed to verify the legitimacy of the server and therefore could not
0.290 establish a secure connection to it. To learn more about this situation and
0.290 how to fix it, please visit the web page mentioned above.
0.293 /bin/sh: 1: pyenv: not found
0.293 Cloning into '/plugins/pyenv-install-latest'...
0.644 /bin/sh: 1: pyenv: not found
------
Dockerfile:33
--------------------
  32 |          && rm -rf /var/lib/apt/lists/*
  33 | >>> RUN curl -s -S -L https://raw.githubusercontent.com/pyenv/pyenv-installer/master/bin/pyenv-installer | bash && \
  34 | >>>      git clone https://github.com/momo-lab/pyenv-install-latest.git "$(pyenv root)"/plugins/pyenv-install-latest && \
  35 | >>>      pyenv install-latest "3.11" && \
  36 | >>>      pyenv global $(pyenv install-latest --print "3.11") && \
  37 | >>>      pip install "wheel<1"
  38 |     COPY .cog/tmp/build2214313725/cog-0.0.1.dev-py3-none-any.whl /tmp/cog-0.0.1.dev-py3-none-any.whl
--------------------
ERROR: failed to solve: process "/bin/sh -c curl -s -S -L https://raw.githubusercontent.com/pyenv/pyenv-installer/master/bin/pyenv-installer | bash && \tgit clone https://github.com/momo-lab/pyenv-install-latest.git \"$(pyenv root)\"/plugins/pyenv-install-latest && \tpyenv install-latest \"3.11\" && \tpyenv global $(pyenv install-latest --print \"3.11\") && \tpip install \"wheel<1\"" did not complete successfully: exit code: 127
ⅹ Failed to build Docker image: exit status 1
```

**Comment by mattt** [comment]
Hi @blevlabs, @voodoohop, @isouf. This appears to be a temporary problem with Ubuntu's distribution of curl.

<cite>From https://stackoverflow.com/a/76724711/157142:</cite>
> 
> There was a broken backport for a security fix which was introduced in an update to curl a few days ago. This bug is about certificate validation when wildcard certificates are used - thus relevant when checking download.docker.com but not when checking google.com.
>
> Should be fixed in 7.81.0-1ubuntu1.13 which fixes 7.81.0-1ubuntu1.11. From https://launchpad.net/ubuntu/jammy/+source/curl/+changelog:

If you try `docker prune` and run `cog build` with `--no-cache`, that should fix the issue.

For context, this issue only presented itself when the `gpu` config option was set to `true`, because that causes Cog to use one of [Nvidia's CUDA base images](https://hub.docker.com/r/nvidia/cuda), which doesn't include Python; we use curl to download pyenv, and use pyenv to download the latest specified Python. On non-GPU images, we use one of the [official Python slim images](https://hub.docker.com/_/python).

[Cog v0.8.2](https://github.com/replicate/cog/releases/tag/v0.8.2), released earlier today, includes an option to build GPU images with a Python base image instead of the CUDA base image (which would also solve the curl issue you encountered). You can opt into that behavior by running  `cog build --use-cuda-base-image=false`.

Going forward, I think we should pin system packages in `apt-get install`, which would get us closer to reproducible builds and help prevent things from randomly breaking over time. See https://github.com/replicate/cog/pull/570#issuecomment-1625951313 for more discussion.

I hope that helps! Let me know if you're still seeing this issue after trying the suggested solutions.

**Comment by iamargentum** [comment]
> If you try docker prune and run cog build with --no-cache, that should fix the issue.
this worked for me
